engine/src/main/java/io/camunda/zeebe/engine/state/appliers/EventAppliers.java

```diff
@@ -64,6 +64,7 @@ public EventAppliers(final MutableProcessingState state) {
     register(ProcessInstanceResultIntent.COMPLETED, NOOP_EVENT_APPLIER);
 
     register(ProcessIntent.CREATED, new ProcessCreatedApplier(state));
+    register(ProcessIntent.DELETED, NOOP_EVENT_APPLIER);
```

Comment:
❌ This behavior would differ with later versions, where an applier is used. So on rolling upgrade, followers on 8.2 would not be able to produce the same state on process deletion. Process deletion is not supported on 8.2, so this should fail rather than noop.

Reply:
Hm, I guess that makes sense :+1: 
I'd have to special-case this intent so that our tests don't complain that there is an intent without a matching applier here.

Reply:
🔧 I'd suggest to add another event applier that throws an exception instead of using the noop applier.

Reply:
See 6d61311

Reply:
> 🔧 I'd suggest to add another event applier that throws an exception instead of using the noop applier.

Why this instead of using the new default behavior of throwing when no event applier is found?

Reply:
😅 Ah, you're right, that's better